src/Diff/Types.hs

```diff
@@ -0,0 +1,199 @@
+
+{-# LANGUAGE DuplicateRecordFields      #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+
+module Diff.Types where
+
+import Data.List.NonEmpty (NonEmpty(..))
+import Data.Map (Map)
+import Data.Set (Set)
+import Data.Text (Text)
+import Nix.Derivation (DerivationOutput (..))
+
+import qualified Patience
+import Data.Aeson
+import GHC.Generics (Generic)
+import Data.Foldable
+import Data.Aeson.Types (listValue)
+
+data Changed a = Changed { before :: a, now :: a }
+
+instance ToJSON a => ToJSON (Changed a) where
+  toJSON = changedToJSON toJSON
+
+newtype TextDiff f = TextDiff {unTextDiff :: f (Patience.Item Text)}
+
+instance Foldable f => ToJSON (TextDiff f) where
+  toJSON = listValue toJSONItem . squashDiff . toList . unTextDiff
+    where
+      -- We want to have the longest difference spans,
+      -- instead of printing each word as one item
+      squashDiff (Patience.Old a : Patience.Old b : xs) =
```

Comment:
Will this squashing affect the textual representation of the given `TextDiff`?

Reply:
I'm not sure, let's test it

Reply:
As I can see, it doesn't affect in NotTTY, and it may affect on IsTTY (colored output)

Reply:
Oh, and it will affect on arguments output, so we should remove this squashing for them and test sources/environment

Reply:
This is quite weird. I'd expect that the deserialized value to act the same way as value before serialization